Fabricator.Resources/HostFile.fs

```diff
@@ -0,0 +1,141 @@
+// SPDX-FileCopyrightText: 2020-2025 Friedrich von Never <friedrich@fornever.me>
+//
+// SPDX-License-Identifier: MIT
+
+namespace Fabricator.Resources
+
+open System
+open System.IO
+open System.Runtime.InteropServices
+open Fabricator.Core
+open TruePath
+open TruePath.SystemIo
+
+type HostFile =
+    /// <summary>
+    /// Gets the default hosts file path for the current platform.
+    /// </summary>
+    static member internal DefaultHostsPath: AbsolutePath =
+        if RuntimeInformation.IsOSPlatform(OSPlatform.Windows) then
+            AbsolutePath @"C:\Windows\System32\drivers\etc\hosts"
+        else
+            AbsolutePath "/etc/hosts"
+    
+    /// <summary>
+    /// Creates a resource for managing a host file entry.
+    /// </summary>
+    /// <param name="ipAddress">The IP address for the host entry.</param>
+    /// <param name="host">The hostname to map to the IP address.</param>
+    /// <param name="hostsFilePath">Optional path to the hosts file. Defaults to the platform-specific hosts file path:
+    /// Windows: C:\Windows\System32\drivers\etc\hosts, Linux/macOS: /etc/hosts</param>
+    /// <returns>An IResource for managing the host file entry.</returns>
+    static member Record(ipAddress: string, host: string, ?hostsFilePath: AbsolutePath): IResource =
+        let filePath = defaultArg hostsFilePath HostFile.DefaultHostsPath
+        
+        let removeInlineComment (line: string) =
+            // Remove inline comments (# til the end of the line)
+            let commentIndex = line.IndexOf('#')
+            if commentIndex >= 0 then
+                line.Substring(0, commentIndex)
+            else
+                line
+        
+        let parseEntry (line: string) =
+            // Parse a line and return (IP, [hosts])
+            // Handles multi-host entries like "192.168.1.1 host1 host2 host3"
+            let lineWithoutComment = removeInlineComment line
+            let parts = lineWithoutComment.Split([|' '; '\t'|], StringSplitOptions.RemoveEmptyEntries)
+            if parts.Length >= 2 then
+                let ip = parts.[0].Trim().ToLowerInvariant()
+                let hosts = parts.[1..] |> Array.map (fun h -> h.Trim().ToLowerInvariant())
+                Some (ip, hosts)
+            else
+                None
+        
+        let expectedHost = host.Trim().ToLowerInvariant()
+        let expectedIp = ipAddress.Trim().ToLowerInvariant()
+        
+        let findEntryWithHost (lines: string[]) =
+            // Find the line that contains our host
+            lines
+            |> Array.tryFindIndex (fun line ->
+                let trimmed = line.Trim()
```

Comment:
We don't need this check for comment anymore, do we? I'd prefer to only have a single place dealing with comments, the one in `parseEntry`.